.github/workflows/test.yaml

```diff
@@ -42,6 +42,9 @@ jobs:
       - name: Run tests
         run: yarn test --silent --coverage
 
+      - name: Run E2E tests on Next.js server
```

Comment:
Do we want to run these E2E tests with a [matrix strategy](https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix) (since both express and next-server should work) ?

Reply:
And maybe split E2E runs from unit runs? Or is there a specific reason we should only run the next server for E2E and use express for the whole thing?

Reply:
If we use the matrix strategy the unit tests will run for the second time for nothing, since they don't differ per server.

---

If we split running unit runs from E2E runs, the coverage will drop since some of the functions are only called via the E2E tests. If we do split in the end, we need to write separate tests to cover the areas that are currently covered by E2E tests.

Reply:
Codecov combines coverage reports, but it does require a few extra steps so let's handle that seperately if we decide to seperate E2E. 

I think if we do decide to run E2E seperately I think the matrix would make sense. 

But let's handle that outside of this PR if we decide to go with that.